radixdlt-java/src/main/java/com/radixdlt/client/application/translate/tokenclasses/TokenTypesReducer.java

```diff
@@ -58,23 +54,28 @@ public TokenClassesState reduce(TokenClassesState state, SpunParticle s) {
 			}
 
 			return state.mergeTokenClass(
-				tokenParticle.getTokenClassReference(),
+				tokenParticle.getTokenTypeReference(),
 				tokenParticle.getName(),
 				tokenParticle.getSymbol(),
 				tokenParticle.getDescription(),
-				TokenClassReference.subunitsToUnits(tokenParticle.getGranularity()),
+				TokenTypeReference.subunitsToUnits(tokenParticle.getGranularity()),
 				tokenSupplyType
 			);
-		} else {
-			OwnedTokensParticle mintedOrBurned = (OwnedTokensParticle) p;
-			BigInteger mintedOrBurnedAmount = UInt256s.toBigInteger(mintedOrBurned.getAmount());
-			BigDecimal change = TokenClassReference.subunitsToUnits(
-				(mintedOrBurned.getType() == FungibleType.BURNED)
-					? mintedOrBurnedAmount.negate()
-					: mintedOrBurnedAmount
-			);
+		} else if (s.getSpin() == Spin.UP) {
+			if (p instanceof MintedTokensParticle || p instanceof BurnedTokensParticle) {
```

Comment:
Would rather have all particle class checks occur in the same "level" e.g.:
```
if (p instanceof TokenParticle) {
...
} else if (p instanceof MintedTokensParticle || p instanceof BurnedTokensParticle) {
    if (s.getSpin() == Spin.UP) {
...
    }
}
```

Reply:
Ha, I specifically didn't nest it out because I like it better that way. But alas, I'm outvoted..